packages/bff/src/services/purposeService.ts

```diff
@@ -24,31 +265,126 @@ export function purposeServiceBuilder(
-        )
-        .map((v) => unsafeBrandId<PurposeVersionId>(v.id))
-        .pop();
+      const versionId = getCurrentVersion(updatedPurpose.versions)?.id;
 
       if (versionId === undefined) {
         throw purposeNotFound(id);
       }
 
-      return { purposeId: id, versionId };
+      return { purposeId: id, versionId: unsafeBrandId(versionId) };
+    },
+    async getProducerPurposes(
+      filters: {
+        name?: string | undefined;
+        eservicesIds?: string[] | undefined;
+        consumersIds?: string[] | undefined;
+        producersIds?: string[] | undefined;
+        states?: purposeApi.PurposeVersionState[] | undefined;
+      },
+      offset: number,
+      limit: number,
+      { headers, authData, logger }: WithLogger<BffAppContext>
+    ): Promise<bffApi.Purposes> {
+      logger.info(
+        `Retrieving Purposes for name ${filters.name}, EServices ${filters.eservicesIds}, Producers ${filters.producersIds}, offset ${offset}, limit ${limit}`
+      );
+      return await getPurposes(
+        {
+          purposeClient,
+          eserviceClient,
+          tenantClient,
+          agreementClient,
+          authorizationClient,
+        },
+        authData.organizationId,
+        { ...filters, excludeDraft: true },
+        offset,
+        limit,
+        headers
+      );
+    },
+    async getConsumerPurposes(
+      filters: {
+        name?: string | undefined;
+        eservicesIds?: string[] | undefined;
+        consumersIds?: string[] | undefined;
+        producersIds?: string[] | undefined;
+        states?: purposeApi.PurposeVersionState[] | undefined;
+      },
+      offset: number,
+      limit: number,
+      { headers, authData, logger }: WithLogger<BffAppContext>
+    ): Promise<bffApi.Purposes> {
+      logger.info(
+        `Retrieving Purposes for name ${filters.name}, EServices ${filters.eservicesIds}, Consumers ${filters.consumersIds}, offset ${offset}, limit ${limit}`
+      );
+      return await getPurposes(
+        {
+          purposeClient,
+          eserviceClient,
+          tenantClient,
+          agreementClient,
+          authorizationClient,
+        },
+        authData.organizationId,
+        { ...filters, excludeDraft: false },
+        offset,
+        limit,
+        headers
+      );
+    },
+    async clonePurpose(
+      purposeId: PurposeId,
+      seed: purposeApi.PurposeCloneSeed,
+      { headers, logger }: WithLogger<BffAppContext>
+    ): Promise<bffApi.PurposeVersionResource> {
+      logger.info(`Cloning purpose ${purposeId}`);
+
+      const cloned = await purposeClient.clonePurpose(seed, {
+        params: {
+          purposeId,
+        },
+        headers,
+      });
+
+      const draft = cloned.versions.find(
+        (v) => v.state === purposeApi.PurposeVersionState.Values.DRAFT
+      );
+      if (!draft) {
+        throw purposeDraftVersionNotFound(purposeId);
+      }
```

Comment:
If convenient for others function it could be a separate function

Reply:
Right now I didn't find any other endpoint that uses this check